### ISSUE TITLE
PowerHandler Enhancement.

### ISSUE THREAD

**Issue by LordFokas** [BuildCraft/BuildCraft#1288]
I'll try to keep this short, but I think it's trivial enough for me to express myself easily.

Currently the PowerHandler uses floats to keep track of stuff.
While this is good enough for numbers with only a few digits (say, the quarry only stores 15k energy), for high values (which is my case) that may cause inaccuracy, with side effects like receiving or losing more energy than was specified.

To sum it up, I'm suggesting using doubles inside PowerHandlers.
This would make the PowerHandler nicer to everyone who's storing more that a few thousands MJ.

Thanks in advance.
